src/tm_devices/helpers/logging.py

```diff
@@ -0,0 +1,158 @@
+# pyright: reportUnnecessaryTypeIgnoreComment=none
+"""Helpers for logging."""
+
+from __future__ import annotations
+
+import importlib.metadata
+import logging
+import sys
+import time
+
+from pathlib import Path
+from typing import Optional, TYPE_CHECKING, Union
+
+import colorlog
+import pyvisa
+
+from tzlocal import (
+    get_localzone,  # pyright: ignore[reportUnknownVariableType]
+)
+
+from tm_devices.helpers.enums import CustomStrEnum
+
+if TYPE_CHECKING:
+    import os
+
+_logger_initialized = False
+
+
+class _CustomFormatter(logging.Formatter):  # pragma: no cover
+    def format(self, record: logging.LogRecord) -> str:
+        # Add the package name to the log record
+        record.package_name = record.name.split(".", maxsplit=1)[0]
+        # Call the original format method
+        return super().format(record)
+
+
+class LoggingLevels(CustomStrEnum):
```

Comment:
Is this mainly for better type hinting for the log levels rather than just using logging.DEBUG, ....

Reply:
It allows an enum to be used that can disable logging with `LoggingLevels.NONE` the `logging` library doesn't easily have that.